improver/calibration/dataframe_utilities.py

```diff
@@ -510,21 +594,39 @@ def truth_dataframe_to_cube(df: DataFrame, training_dates: DatetimeIndex,) -> Cu
         time_coord = _define_time_coord(adate, time_bounds)
         height_coord = _define_height_coord(time_df["height"].values[0])
 
-        cube = build_spotdata_cube(
-            time_df["ob_value"].astype(np.float32),
-            time_df["cf_name"].values[0],
-            time_df["units"].values[0],
-            time_df["altitude"].astype(np.float32),
-            time_df["latitude"].astype(np.float32),
-            time_df["longitude"].astype(np.float32),
-            time_df["wmo_id"].values.astype("U5"),
-            scalar_coords=[time_coord, height_coord],
-        )
-        cubelist.append(cube)
+        if var_type == "threshold":
+            for var_val in sorted(df[var_type].unique()):
+                var_coord = DimCoord(np.float32(var_val), long_name=var_type, units=1)
+                var_df = time_df.loc[time_df[var_type] == var_val]
+                cube = build_spotdata_cube(
+                    (var_df["ob_value"] > var_val).astype(np.int32),
```

Comment:
I've added this option.

Reply:
Hi Both, For analyses/fcts etc this thresholding step is a separate command? Should we be moving this functionality to that part of the code? Or should we leave this in here?

Reply:
Thanks for the query, @anja-bom. Ideally, I think that it would be better if we could avoid the data transformation (i.e. the thresholding) when doing the conversion from dataframes to cubes, just to keep the functionality cleaner. You're correct that there is already a separate command for thresholding, so it would make sense and likely be more maintainable, if we could re-use that functionality. In this case, I would think that the approach would be to just create a truth cube (as already done for realizations and percentiles) and then threshold later, outside of `dataframe_utilities.py`. Would that make sense?

Reply:
I've removed the code to threshold the truth data. I tested using the improver threshold cli as a subsequent step, which seems to work fine.